common/src/main/java/com/zegelin/cassandra/exporter/collector/StorageServiceMBeanMetricFamilyCollector.java

```diff
@@ -76,6 +79,30 @@ private StorageServiceMBeanMetricFamilyCollector(final StorageServiceMBean stora
         this.labeledFileStores = ImmutableMap.copyOf(labeledFileStores);
     }
 
+    private Labels decodeTokenRange(String tokenRange, String localIP, String keyspace) {
+        HashMap<String, String> m = Maps.newHashMap(metadataFactory.endpointLabels(localIP));
+        HashSet<String> endpoints = new HashSet<String>();
+
+        // token range example:
+        // TokenRange(start_token:5585272669612250202, end_token:5664918566912044362, endpoints:[172.16.28.48, 172.16.28.166], rpc_endpoints:[172.16.28.48, 172.16.28.166], endpoint_details:[EndpointDetails(host:172.16.28.48, datacenter:eu-west_edge-irl1_profiles-bk, rack:1a), EndpointDetails(host:172.16.28.166, datacenter:eu-west_edge-irl1_profiles-bk, rack:1c)])
+        // see https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/service/TokenRange.java
+        String s = tokenRange.substring(tokenRange.indexOf(", endpoints:") + 13);
```

Comment:
Are the token ranges always always in valid format? If not, should we have a regex to validate them before going ahead with these splits/substrings?

Reply:
The format is fixed, and it's very unlikely to change in the future. It seems to have been added just to preserve some backward compatibility, and it's unlikely to ever change. It's used only by a nodetool option, `nodedetool describering`. Anyway, I'll change it to a regex match and group, it's probably safer anyway.